File: leetcode/sliding-window/0209_min_size_subarray_sum.py
# LC 209. Minimum Size Subarray Sum | Medium
class Solution:
    # Sliding window - O(n)
    def minSubArrayLen(self, target, nums):
        l, s, res = 0, 0, float('inf')
        for r in range(len(nums)):
            s += nums[r]
            while s >= target:
                res = min(res, r - l + 1)
                s -= nums[l]; l += 1
        return res if res != float('inf') else 0

File: leetcode/sliding-window/0209_min_size_subarray_sum.cpp
// LC 209. Minimum Size Subarray Sum | Medium
#include <vector>
#include <climits>
using namespace std;

class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int l = 0, s = 0, res = INT_MAX;
        for (int r = 0; r < nums.size(); r++) {
            s += nums[r];
            while (s >= target) { res = min(res, r - l + 1); s -= nums[l++]; }
        }
        return res == INT_MAX ? 0 : res;
    }
};
